pandas/tests/typing/README.md

```diff
@@ -0,0 +1,36 @@
+## Purpose of those tests
+
+The tests contained in the `valid` directory are snippets that when
+process through a type checker ensure that type annotations and type
+stubs from this repository conform to common pandas API use-patterns.
+
+## Running the tests
+
+Tests can be run in following ways:
+
+`pyright pandas/tests/typing`
+
+`mypy pandas/tests/typing`
+
+They'll also be automatically detected and executed by pytest. This
+is to ensure that the test code itself is valid.
+
+## Developing the tests
+
+Some tests contain type checker ignore-instructions along with an
+error that's supposed to be thrown.
+
+    # error: No overload variant of "to_datetime" matches argument type "DataFrame"
+    pd.to_datetime(df)  # type: ignore[call-overload]
+
+All such constructs are placed because of the missing/invalid API
+type information. When the API signature becomes valid again type
+checker will ask you to remove `type: ignore`. Please remove the
+above comment as well.
+
+When adding new tests please use the above solution as well.
+
+## Origins and attribution
+
+The tests come from the [pandas-stubs](https://github.com/VirtusLab/pandas-stubs)
+repository originally released under the MIT license.
```

Comment:
@shoyer You seem to often reply to license compatibility questions :)

Is it okay to integrate [this MIT-licened code](https://github.com/VirtusLab/pandas-stubs/tree/master/tests/snippets) into pandas?

Reply:
Pandas is happy to reuse MIT licensed code (e.g., see existing examples in https://github.com/pandas-dev/pandas/tree/main/LICENSES), but you absolutely need to preserve original copyright notices. The general rule (not sure if pandas is 100% strict about it, but it's a good practice) is to keep the original copyright notice as a top level comment inside each file.

Reply:
@shoyer Should I include any open source contributors, who added their code since the library was originally created, in the copyright notice?